radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/RadixEngineStateComputer.java

```diff
@@ -227,8 +227,10 @@ public void addToMempool(MempoolAdd mempoolAdd, @Nullable BFTNode origin) {
             txn -> {
               try {
                 addToMempool(txn, origin);
+              } catch (MempoolDuplicateException ex) {
+                log.trace("Transaction {} was not added as it was already in the mempool", txn.getId());
```

Comment:
don't we need the cause here?

Reply:
The cause just says that a duplicate transaction was there, which causes loads of useless repetition and stack traces in logs. We have all the information we need here.

Reply:
So maybe we should log the original message or change the original message to the one created now?

Reply:
For me, there's an abstraction difference.

The exception is thrown in a place where the mempool add could have come via different paths (eg from a full node adding it themselves due to a submission, or due to a validator adding a vote transaction) - these may want to handle the exception differently, and so the message should be quite general, and describe the problem. EG if someone submits a transaction from the Core API, we'd want to report it's a duplicate; but if it comes from Node Gossip (as in this case),  that's totally expected.

In this particular catch, we know that the message is from Node Gossip, and so we're fine to discard duplicates. The message should reflect that we've discarded a duplicate and we're not worried.